### PR TITLE
Remove CA2010 (ConsumePreserveSigAnalyzer)

### PR DESCRIPTION
**Pull request by mavasani**
1. This analyzer has no unit tests
2. Its functionality is a subset of functionality provided for [CA1806: Do not ignore method results](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1806)
3. The functionality provided in this analyzer is not completely correct: See #1864. #746 tracks implementing this functionality correctly in CA1806 analyzer
4. This analyzer has no documentation: https://docs.microsoft.com/visualstudio/code-quality/ca2010 gives 404

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.sarif [hunk 1]
```diff
@@ -2457,6 +2457,25 @@
             ]
           }
         },
+        "CA1835": {
+          "id": "CA1835",
+          "shortDescription": "Prefer 'ReadOnlyMemory<Byte>' overload for 'WriteAsync'.",
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.sarif:2462]: @carlossanlop I think you may want to update the Title and Description for CA1835 rules to be shared, i.e. something like `Prefer 'ReadOnlyMemory<Byte>' overload for 'ReadAsync'/'WriteAsync'` and use it for both your descriptors.

**Comment by mavasani** [comment]
Ping - trivial review deleting code.

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.md [hunk 2]
```diff
@@ -103,12 +103,12 @@ Rule ID | Title | Category | Enabled | Severity | CodeFix | Description |
...
 [CA1834](https://docs.microsoft.com/visualstudio/code-quality/ca1834) | Consider using 'StringBuilder.Append(char)' when applicable. | Performance | True | Warning | True | 'StringBuilder.Append(char)' is more efficient than 'StringBuilder.Append(string)' when the string is a single character. When calling 'Append' with a constant, prefer using a constant char rather than a constant string containing one character. |
+[CA1835](https://docs.microsoft.com/visualstudio/code-quality/ca1835) | Prefer 'ReadOnlyMemory<Byte>' overload for 'WriteAsync'. | Performance | True | Warning | False | All 'Stream' based classes provide a 'WriteAsync' overload that takes a 'ReadOnlyMemory<Byte>' as first argument. When possible, prefer the overload that takes a 'ReadOnlyMemory<Byte>', which is more efficient. |
```
Comment (stephentoub) [inline src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.md:106]: Why is this being added in this PR?  Was it just missing erroneously?

File: src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt [hunk 3]
```diff
@@ -17,7 +17,7 @@ Security: CA2100-CA2153, CA2300-CA2330, CA3000-CA3147, CA5300-CA5403
 Usage: CA1801, CA1806, CA1816, CA2200-CA2209, CA2211-CA2247
 Naming: CA1700-CA1726
 Maintainability: CA1500-CA1509
-Reliability: CA9999, CA2000-CA2015
+Reliability: CA9999, CA2000-CA2009, CA2011-CA2015
```
Comment (stephentoub) [inline src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt:20]: Are we ok reusing CA2010 for something else in the future?

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.md (hunk 2 above)
Comment (mavasani) [reply src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.md:106]: This was me just running `/t:pack` locally, which runs the tool to auto-update the SARIF and md files to reflect the current state. I can revert these changes and have them done in a separate cleanup PR later on.

File: src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt (hunk 3 above)
Comment (mavasani) [reply src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt:20]: Historically, we have avoided repurposing IDs given users might have existing suppressions/ruleset entries to turn these off. I am in two minds if we should make this change to indicate that we no longer use CA2010 OR keep the original range so we don't accidentally pick up CA2010. Maybe safer to go with the latter, let me revert this change.

**Comment by mavasani** [comment]
Thanks - that made the PR much cleaner and only has deletions now.

**Pull request merged**